src/utils/liveTickerTooltip.js

```diff
@@ -0,0 +1,145 @@
+      positionTooltip(event); // initial position
+    }
+
+    return tooltip;
+  }
+
+  function setToolTipVisible(isVisible = true) {
+    const tooltip = document.querySelector('.liveTickerTooltip');
+    if (tooltip) {
+      if (isVisible) {
+        tooltip.classList.remove('hidden');
+      } else {
+        tooltip.classList.add('hidden');
+      }
+    }
+  }
+
+  function getTickersFromElement(tickerElement) {
+    let matches;
+
+    const TICKER_REGEX = /\b[A-Z]{1,6}\b/g; // do we need to include numbers?
+    const ignoredTextContents = [
+      'Cash Remainder',
+      'US Dollar',
+    ];
+
+    const parentElement = tickerElement?.closest(tickerParentSelectors.join(', '));
+    if (parentElement) {
+      const isIgnored = ignoredTextContents.some(
+        text => tickerElement?.textContent?.match(text, 'gi')
+      );
+      if (!isIgnored) {
+        matches = tickerElement?.textContent?.match(TICKER_REGEX);
+      }
+    }
+    return matches ? matches?.map(match => match.trim().toUpperCase()) : [];
+  }
+
+  const debouncedShowTooltip = _.debounce(function showTooltip(ticker, event) {
+    if (!ticker) {
+      setToolTipVisible(false);
+      return;
+    }
+
+    // Create and position the new tooltip
+    const tooltip = createOrGetTooltip(event);
+    setToolTipVisible(true);
+    tooltip.innerHTML = '<div class="spinner"></div>'; // Spinner while loading
+
+    // Prepare the data to send in the request body
+    const requestData = {
+      tickers: [
+        `EQUITIES::${ticker}//USD`
+      ]
+    };
+
+    let dataPromise;
+    if (tickerCache[ticker]) {
+      dataPromise = Promise.resolve(tickerCache[ticker]);
+    } else {
+      dataPromise = fetch(API_URL, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(requestData)
+      }).then(
+        response => response.json()
+      );
+    }
+
+    dataPromise.then(data => {
+      const filteredData = Object.entries(data).filter(
+        ([key, value]) => key !== `$USD`
+      ).map(([key, value]) => {
+        return {...value, key}
+      });
+      let name = filteredData[0]?.name;
+
+      // Cache the data
+      tickerCache[ticker] = filteredData;
+      localStorage.setItem('tickerCache', JSON.stringify(tickerCache));
+
+      // if the first item is USD, then the requested ticker was not found
+      if (!name) {
+        setToolTipVisible(false);
+      } else {
+        tooltip.innerHTML = `<div class="tooltip-content">${ticker}: ${name}</div>`;
+      }
+    }).catch(error => {
+      tooltip.innerHTML = `<div class="tooltip-content">${ticker}: Error loading info</div>`;
+    });
+  }, 500); // should be long enough for content to load so that cache can be used
+
+  const debouncedCheckTicker = _.debounce(function(event) {
+    const elementFromPoint = document.elementFromPoint(event.clientX, event.clientY);
```

Comment:
oh thats cool. I didnt know about elementFromPoint

Reply:
Yah, it’s not perfect but works ok for our use case. I went with this because there isn’t a great way to select tickers otherwise.